kyaml/yaml/fns_test.go

```diff
@@ -57,6 +57,47 @@ func TestAppend(t *testing.T) {
 	assert.Nil(t, rn)
 }
 
+func TestGetElementByKey(t *testing.T) {
+	node, err := Parse(`
+- b: c
+- i
+- d: e
+- f: g
+- f: h
+`)
+	assert.NoError(t, err)
+
+	rn, err := node.Pipe(GetElementByKey("b"))
+	assert.NoError(t, err)
+	assert.Equal(t, "b: c\n", assertNoErrorString(t)(rn.String()))
+
+	rn, err = node.Pipe(GetElementByKey("f"))
+	assert.NoError(t, err)
+	assert.Equal(t, "f: g\n", assertNoErrorString(t)(rn.String()))
+}
+
+func TestElementMatcherWithNoValue(t *testing.T) {
+	node, err := Parse(`
+- a: c
+- b: ""
+`)
+	assert.NoError(t, err)
+
+	rn, err := node.Pipe(ElementMatcher{FieldName: "b"})
+	assert.NoError(t, err)
+	assert.Equal(t, "b: \"\"\n", assertNoErrorString(t)(rn.String()))
+
+	rn, err = node.Pipe(ElementMatcher{FieldName: "a"})
+	assert.NoError(t, err)
+	assert.Equal(t, "", assertNoErrorString(t)(rn.String()))
```

Comment:
L92 is meant to demonstrate a match failure for fieldValue "".

The documentation for `type ElementMatcher struct` doesn't actually specify
what is returned if there's no match- it's nil, but we should say so up above.

So L92 is technically correct, since (nil *RNode).String returns "" , 
but lets make it just
```
assert.Nil(t, rn)
```